promql/engine.go

```diff
@@ -1267,7 +1271,10 @@ func (ev *evaluator) eval(expr parser.Expr) (parser.Value, storage.Warnings) {
 			}
 		}
 
-		sel := unwrapStepInvariantExpr(e.Args[matrixArgIndex]).(*parser.MatrixSelector)
+		unwrapParenExpr(&e.Args[matrixArgIndex])
+		arg := unwrapStepInvariantExpr(e.Args[matrixArgIndex])
+		unwrapParenExpr(&arg)
```

Comment:
can this `unwrapParenExpr` be inside `unwrapStepInvariantExpr`?

Reply:
We could do that, but the name has to be updated

Reply:
I tried to do this (with function renamed to `unwrapStepInvariantAndParenExpr`), but looks like I don't have permissions to push to this branch. Merging this for now, feel free to open a PR with this change :) Thanks!